src/twisted/internet/defer.py

```diff
@@ -160,6 +166,65 @@ def maybeDeferred(f, *args, **kw):
 
 
 
+def unbox(deferred, raises=True, passthrough=True):
```

Comment:
``extractImmediateResult()`` is kind of long, what about just ``extractResult``? Or if that doesn't convey the meaning well enough, what about ``extractCurrentResult()`` which I think is clearer that it's what the current state is, not some future state and not "block and get a result".